ext/hook/uhook.stub.php

```diff
@@ -145,5 +152,7 @@ function install_hook(
  * @param int $id The id to remove.
  * @param string $location A class name (which inherits this hook through inheritance), which to specifically remove
  * this hook from.
+ * @return void no return, not formally declared void because of a buggy debug assertion in PHP 7.1
+ *              ("return value must be of type void, null returned")
  */
-function remove_hook(int $id, string $location = ""): void {}
+function remove_hook(int $id, string $location = "") {}
```

Comment:
Why would that happen? https://heap.space/xref/PHP-7.1/Zend/zend_execute.c?r=7f6387b5#1045 seems to properly check for it?

Reply:
The error is not in the condition that you point out. That check does indeed pass. The problem is that it's missing a return so it fails the check next. See in PHP 7.2: https://heap.space/xref/PHP-7.2/Zend/zend_execute.c?r=902d39a3#998

Reply:
Your link doesn't help - because you show a fixed version.
https://github.com/php/php-src/commit/9b0df97a42749464069859c5ca77c0241a8b05ca is likely what you mean - all versions before 7.1.20 are broken.

Reply:
Yeah I was just pointing to return statement that was added later.